src/packages.rs

```diff
@@ -0,0 +1,119 @@
+use serde::{Deserialize, Serialize};
+
+use std::process::Command;
+
+use crate::Config;
+use crate::Module;
+
+#[derive(Deserialize, Serialize)]
+#[serde(default)]
+pub struct Packages {
+    #[serde(skip)]
+    package_counts: Vec<PackageCount>,
+
+    header: String,
+    /// Whether to show the package manager names. Format: "count (pkg_manager)"
+    show_package_managers: bool,
+}
+
+struct PackageCount {
+    count: u32,
+    package_manager: String,
+}
+
+impl PackageCount {
+    fn new(count: u32, package_manager: &str) -> Self {
+        Self {
+            count,
+            package_manager: String::from(package_manager),
+        }
+    }
+}
+
+struct PackageManager<'a> {
+    name: &'a str,
+    args: &'a [&'a str],
+}
+
+impl<'a> PackageManager<'a> {
+    const fn new(name: &'a str, args: &'a [&'a str]) -> Self {
+        Self { name, args }
+    }
+
+    fn is_installed(&self) -> bool {
+        Command::new(self.name).output().is_ok()
+    }
+
+    fn get_package_count(&self) -> Option<PackageCount> {
+        let command_output = Command::new(self.name).args(self.args).output();
+        let package_list = match command_output {
+            Ok(output) => output.stdout,
+            Err(e) => {
+                eprintln!("Failed while getting package count, {}", e);
+                return None;
+            }
+        };
+        let count = String::from_utf8_lossy(&package_list).lines().count() as u32;
+        Some(PackageCount::new(count, self.name))
+    }
+}
+
+const PACKAGE_MANAGERS: [PackageManager; 6] = [
```

Comment:
nice change :) i would do `&'static [PackageManager] = &[...]` though so we don't need to keep track of the number of elements

Reply:
i would also consider moving this to the top so it's easy to add new package managers